### ISSUE TITLE
Middle Earth Shadow of Mordor carshes at startup

### ISSUE THREAD

**Issue by ahmed-elsayed2017** [doitsujin/dxvk#457]
System information
GPU: HD 8750M
Driver: Mesa 18.1.1 (Padoka stable)
Wine version: 3.10 staging
DXVK version: 0.54

Middle Earth Shadow of Mordor GOTY crashes at startup.

I don't know what is the status of this game on DXVK. Please, someone confirm if this game works with DXVK or not.

[ShadowOfMordor_d3d11.log](https://github.com/doitsujin/dxvk/files/2136143/ShadowOfMordor_d3d11.log)
[ShadowOfMordor_dxgi.log](https://github.com/doitsujin/dxvk/files/2136144/ShadowOfMordor_dxgi.log)

**Comment by jarrard** [comment]
For starters your using the older version DXVK, and you should try mesa 18.1.2+

Also you should enable fake dx10 support which is what it complains about, that might get it going if your lucky.

**Comment by K0bin** [comment]
The game doesn't actually need D3D10.

Setting ``DXVK_FAKE_DX10_SUPPORT=1`` made it run for me.

**Comment by doitsujin** [comment]
`DXVK_FAKE_DX10_SUPPORT=1` seems to be necessary to work around an error message complaining about missing Dx11 support. Is that what you're getting?

But even with that it doesn't work for me, but it also doesn't work with wined3d - one CPU core is spinning at 100% and nothing is happening, so the game likely requires some messing around with wine, or straight-up won't work.

**Comment by ziabice** [comment]
I confirm the game isn't working on my PC (Mesa 18.1.1, AMD RX 580 4GB).

The Linux port has very poor performance, so trying the game with DXVK will be a valid alternative, IMHO. I get only one third of the FPS I get under Windows.

**Comment by Leopard1907** [comment]
This game has a Linux port anyway. Yes , it performs poorly compared to Windows version but that shouldn't be a problem for an up to date hardware. You can easily hit 60 fps and more with everything maxed.

**Comment by jarrard** [comment]
Yeah without log files its pretty hard to help.

**Comment by doitsujin** [comment]
@jarrard in my experience, the Linux port runs fine on AMD (RX 480 here).

I'd gladly get any bugs with this game fixed but at the moment it appears that it doesn't work with wine *at all*, in which case there's not an awful lot I can do about it.

**Comment by ziabice** [comment]
@doitsujin sorry, I'm not a native english speaker, what do you mean for "specify "isn't working""?

The game works natively under Linux with OpenGL with poor perfomance, but it doesn't at all under wine or wine+dxvk. Last time I tried (last weekend), I deleted the game, so I can't reproduce, if needed, the issue now.

**Comment by Leopard1907** [comment]
@ziabice If a game does not work with Wine at all , he can't do anything about it.

Requirements for opening a issue , with "not working " title:

Game needs to run with Wine first , but it doesn't work with DXVK.

**Comment by ziabice** [comment]
@Leopard1907 You are perfectly right, but I haven't opened this issue

**Comment by Leopard1907** [comment]
@ziabice Yeah ,lol. Sorry for confusion :D

**Comment by doitsujin** [comment]
@ziabice what *exactly* did you mean by this:

> I confirm the game isn't working on my PC

I need more details.

**Comment by ahmed-elsayed2017** [comment]
I need to ask a question if you excuse me : If I run a game that support Vulkan like Doom, do I need to turn off DXVK?

**Comment by Leopard1907** [comment]
That Intel and AMD gpu's are not supported notice is an old issue.

Back in the time , AMD open source was not good and closed source AMD driver fglrx was a literal shit.

That is an old notice. Reasons that game runs poorly on Linux was:

Feral was not that experienced

Port based on OpenGL because there was no Vulkan at that time

Dx to OGL conversion literally blocks optimization. 

For example ; Rise of The Tomb Raider works great and not have that kind of huge performance difference. Because Vulkan gave them much more room for performance. 

TL DR ; AMD gpu's are not supported warning is not relevant now.

**Comment by ziabice** [comment]
> @ziabice what exactly did you mean by this:

> >   I confirm the game isn't working on my PC

> I need more details.

I'm downloading the game right now, I'll reply ASAP

**Comment by ahmed-elsayed2017** [comment]
I tried the native version of Middle Earth the Shadow of Mordor, and it is not working at all.

**Comment by Leopard1907** [comment]
@ahmed Let me guess: It stayed on loading screen with a spinning logo?

**Comment by Leopard1907** [comment]
Just try this:

Go to your Steam library - Right Click Shadow of Mordor - Properties

Set launch options:

LANG=C %command%

**Comment by Ahmed** [comment]
Wrong ahmed

**Comment by Leopard1907** [comment]
@ahmed Sorry :(

**Comment by ziabice** [comment]
@doitsujin here are my tests results:

System specs:
CPU AMD A10-7850K, 16 GB
GPU AMD RX 580 4GB
Distro: Manjaro Linux
Drivers: Mesa 18.1.1, amdgpu
Wine-staging-nine 3.10 (from archlinux)
DXVK 0.60 (from AUR dxvk-bin)

Wine run from PlayOnLinux and the console. Game run from Steam installed into a 64bit prefix.

First run: game refuses to start complaining about missing DirectX 11
Second run: added `DXVK_FAKE_DX10_SUPPORT=1` at launch. The game started (Steam says it is running), but it eats 100% CPU on one single core, stuck. No disk activity.

[ShadowOfMordor_d3d11.log](https://github.com/doitsujin/dxvk/files/2146966/ShadowOfMordor_d3d11.log)
[ShadowOfMordor_dxgi.log](https://github.com/doitsujin/dxvk/files/2146967/ShadowOfMordor_dxgi.log)

Hope this helps!

**Comment by doitsujin** [comment]
@ziabice thanks, that's the same issue that I'm having as well but it might not be a DXVK issue. The same thing also happens with wined3d for me.

Does that work for anyone here?

**Comment by ahmed-elsayed2017** [comment]
@Leopard1907 No, it gives me these errors : 

libGL error: unable to load driver: radeonsi_dri.so
libGL error: driver pointer missing
libGL error: failed to load driver: radeonsi
libGL error: unable to load driver: radeonsi_dri.so
libGL error: driver pointer missing
libGL error: failed to load driver: radeonsi
libGL error: unable to load driver: swrast_dri.so
libGL error: failed to load driver: swrast
X Error of failed request:  BadValue (integer parameter out of range for operation)
  Major opcode of failed request:  155 (GLX)
  Minor opcode of failed request:  3 (X_GLXCreateContext)
  Value in failed request:  0x0
  Serial number of failed request:  97
  Current serial number in output stream:  98

**Comment by ahmed-elsayed2017** [comment]
When I open any game, it uses only 25% of my cpu. Is that ok?

**Comment by Leopard1907** [comment]
@ahmed-elsayed2017 Your driver is not configured correctly.

Use Padoka Ppa if you're on Ubuntu or Ubuntu based distros ( like Mint ) , to update your driver.

https://launchpad.net/~paulo-miguel-dias/+archive/ubuntu/pkppa

**Comment by ahmed-elsayed2017** [comment]
It is already installed.

**Comment by ahmed-elsayed2017** [comment]
@Leopard1907 I fixed the problem with steam on Linux. The native version is not bad. It doesn't give the same framerate as Windows, but it's playable.

**Comment by Leopard1907** [comment]
@ahmed-elsayed2017

Nice. Personally , i only prefer Wine + DXVK for non native titles. Using native versions for other games no matter their performance rates are. Shadow of Mordor would benefit from a Vulkan port like Feral did with Mad Max but asking for a revisit for these older titles would be pointless.

**Comment by K0bin** [comment]
If it doesn't work there's either something wrong with your setup or it's a bug in Wine.

The game runs great with DXVK on Windows.

![image](https://user-images.githubusercontent.com/1131720/44294141-d955bd80-a292-11e8-9a95-25fedd02d7fa.png)

This issue can be closed.

**Comment by pchome** [comment]
@massatt212 
> I think it's caused by the amd spoof for Nvidia

You can check this out by using e.g. https://github.com/simons-public/protonfixes

`util.protontricks('vd=1920x1080')` -- wine virtual desktop
`util.protontricks('force_gpu=nvidia')` -- report Nvidia GPU by wine (Proton)

`util.set_dxvk_option('dxgi.customDeviceId', '10de')` -- 
`util.set_dxvk_option('dxgi.customVendorId', '13c2')` -- report Nvidia GPU by  DXVK

**Comment by evanrush** [comment]
find . \( -name "libgcc_s.so*" -o -name "libstdc++.so*" -o -name "libxcb.so*" -o -name "libgpg-error.so*" \) -print -delete

**Comment by evanrush** [comment]
or "LD_PRELOAD='/usr/$LIB/libstdc++.so.6' DISPLAY=:0 DRI_PRIME=1 sh run_game.sh"
